libr/core/disasm.c

```diff
@@ -3747,12 +3747,18 @@ static void ds_print_ptr(RDisasmState *ds, int len, int idx) {
 					}
 				}
 			}
-			//XXX this should be refactored with along the above
-			kind = r_anal_data_kind (core->anal, refaddr, (const ut8*)msg, len - 1);
+			// XXX this should be refactored with along the above
+			char *m = msg;
+			int l = len;
+			if (l > 1 && !IS_PRINTABLE (*m) && *m < l) {
+				m++;
+				l--;
+			}
+			kind = r_anal_data_kind (core->anal, refaddr, (const ut8*)m, l);
 			if (kind) {
 				if (!strcmp (kind, "text")) {
-					if (!string_printed && *msg) {
-						ds_print_str (ds, msg, len, refaddr);
+					if (!string_printed && *m) {
+						ds_print_str (ds, m, l, refaddr);
```

Comment:
If non-printable chars are skipped, there should be some kind of visual indicator, e.g. `.."str"`

Reply:
Those chars are not part of the string its jusy the length

Reply:
Ok but might cause issues if people expect the string to be at addr and not at addr+n